### ISSUE TITLE
Use latest version from commit history instead latest greatest for unchanged modules

### ISSUE THREAD

**Issue by nitschSB** [danielflower/multi-module-maven-release-plugin#118]
Hello Daniel,
first of all great work! I'm really looking forward to use this plugin. I played a lot with the bundled `dependencymanagement-using-parent-module-version-property` test project and found this issue that is a deal-breaker for us.

The plugin always uses the latest version for unchanged modules (info log: _Will use version [...] as it has not been changed since that release._) but this does not work for changes on different branches.

Imagine this little git example:
```
      C   topic
     /
    A---B master

A: initial commit; tags: util-1.0.1 app-1.0.1
B: change util; tags: util-1.0.2 app-1.0.2
C: change app; build fails (tries to use incompatible util-1.0.2)
```
Each commit gets released and commit B contains a breaking change in a util-module. Commit C only contains a change of the app-module (that uses the util-module). The release of C will fail because it detects no changes on util-module (which is correct) but uses the version build from commit B, which has changes in comparison to C.

I recommend to walk back the history of commit C and only use tags you find in this history as possible version candidates.

In my example the release of C should use util-1.0.1 to build app-1.0.3 and everything would be fine.

What do you think about this issue/suggestion?

**Comment by danielflower** [comment]
Hi there, thanks for the details. Yes, it sounds like what you point out is an issue. I don't remember all the details of this plugin but what you suggested sounds it makes sense.

**Comment by kobynet** [comment]
This is actually an issue we had as well, forcing us to release from a single branch. 
Thank's for the effort @nitschSB !!

**Comment by danielflower** [comment]
Thanks, released in `3.6.0`
